### PR TITLE
feat(Core/Spells): Implemented SPELL_AURA_DETECT_AMORE.

### LINKED ISSUE #3443: (DB/Quest): The Hunt is on [$15]
##### CURRENT BEHAVIOUR:

- [ ] Quest giver gives you the "Oculus of the Exorcist" which allows you to "detect recent users of cultist necromancy" When using the item the following NPCs should have the spell "Shroud of the Death Cultist" cast on them:

NPC_TOM_HEGGER                              = 25827,
NPC_SALTY_JOHN_THORPE                       = 25248,
NPC_GUARD_MITCHELLS                         = 25828,

Even though they do not have "Shroud of the Death Cultist" look to them. You are able to complete the quest. 

- [ ] I also noticed that the NPC Tom Hegger display's the $c in the aggro text "You don't know who you're messing with, $c! Death beckons!"

##### EXPECTED BLIZZLIKE BEHAVIOUR:

After using the "Oculus of the Exorcist" you should see the 3 NPC listed above with the "Shroud of the Death Cultist" look to them. 

For the aggro text of Tom Hegger, the $c should be replaced with the players class.

##### STEPS TO REPRODUCE THE PROBLEM:

1. .tele Borean
2. .quest add 11794
3. use the Oculus of the Exorcist while standing in front of the npc's listed above

##### AC HASH/COMMIT:

https://github.com/azerothcore/azerothcore-wotlk/commit/3934678b07603bb44189f16bebb32a25b19bc23f

##### OPERATING SYSTEM:

Centos 7

##### MODULES:

mod-ah-bot
mod-autobalance
mod-cfbg
mod-custom-login
mod-individual-xp
mod-transmog

##### OTHER CUSTOMIZATIONS:

<bountysource-plugin>

---
There is a **[$15 open bounty](https://www.bountysource.com/issues/92831376-db-quest-the-hunt-is-on?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #3443.

The text problem from this issue is already fixed.

## Changes Proposed:
-  
-  

## Issues Addressed:
- Closes #3443

## Tests Performed:
-
-

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -1671,6 +1671,32 @@ void AuraEffect::HandleModStealthLevel(AuraApplication const* aurApp, uint8 mode
     target->UpdateObjectVisibility(target->GetTypeId() == TYPEID_PLAYER || IS_PLAYER_GUID(target->GetOwnerGUID()));
 }
 
+void AuraEffect::HandleDetectAmore(AuraApplication const* aurApp, uint8 mode, bool apply) const
+{
+    if (!(mode & AURA_EFFECT_HANDLE_SEND_FOR_CLIENT_MASK))
+        return;
```
Comment (FrancescoBorzi) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1677]: same note here: brackets, here and below. please add :)

**Comment by pklloveyou** [comment]
Tested

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by UltraNix** [comment]
Updated.

**Comment by FrancescoBorzi** [comment]
@UltraNix no need to update it constantly, we will take care of it :)

just FYI, this is our merge process: https://www.azerothcore.org/wiki/merge-process

BUT.... if you need some PRs to be merged faster, just let us know and we can merge them sooner

**Pull request merged**